design/one-pager-template-stacks.md

```diff
@@ -0,0 +1,344 @@
+
+#### Concerning the use of Core Types
+
+In several examples throughout this design, Template Stacks are demonstrated generating `Deployment` resources by including them in template text. There are compelling justifications for both permitting and denying Template Stacks from declaring a dependency on core and extension types, such as `deployment.apps/v1`.
+
+The ability to use Template Stacks to render arbitrary Kubernetes resources is a powerful feature. Such a privilege would change the boundaries of a Stack, permitting it to function outside of Crossplane managed types. Template Stacks would compete with `KubernetesApplication` on this capability.
+
+If Template Stacks are prohibited from using core and extension types, the complement of `KubernetesApplication` and a local `KubernetesCluster` can be used to overcome the limitation while providing additional security boundaries. These are features and challenges to investigate in a future design.
+
+### Life Cycle and Updates
+
+During creation and after reconciling, the SM should annotate the Service Account and Deployment to represent the latest version of the Stack.
+
+What should happen if the `Deployment` is manually deleted? Should the Stack Manager recreate it? This is a question left for future designs.
+
+*What happens when a Template Stack is updated? Let's consider how RBAC rules changed handled when the owned or dependent types change. Roles are checked server side on each API call, so a set of Role updates that the Stack Manager may issue to update the Service Account assigned to a Stack will take affect immediately. Unless the deployment is stopped during updates, race conditions could permit template rendered resources more or less roles than needed. If Template Stacks are made to only operate on Crossplane Managed kinds, this may be less concerning. Future designs should explore the effect of role changes on Template Stacks that can create core and extension types.*
+
+#### Unknown Types
+
+Hand-crafted controllers have the advantage of knowing the types and structure of the resources they manage. In Go, this allows for the advantages of a strongly typed environment including compile time error checking and code completion during development.
+
+A Template Stack would not include a controller. An external controller will have to handle reconciliation for types that the controller could not have been aware of at compile time.
+
+How can an external controller manage arbitrary types declared at runtime?
+
+```go
+var obj unstructured.Unstructured
+obj.SetAPIVersion("foo/v1")
+obj.SetKind("Bar")
```

Comment:
`foo/v1` and `Bar` has no effect but need not to be empty, right?

Reply:
Correct, the type will be taken at runtime by examining the `Stack` resource that this TSM instance is operating against.